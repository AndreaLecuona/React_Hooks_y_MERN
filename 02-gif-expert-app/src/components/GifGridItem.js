import React from 'react';
import Proptypes from "prop-types";

export const GifGridItem = ( { id, title, url } ) => {

    // console.log(id, title, url);

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
};

GifGridItem.proptypes = {
    title: Proptypes.string.isRequired,
    url: Proptypes.string.isRequired
};
