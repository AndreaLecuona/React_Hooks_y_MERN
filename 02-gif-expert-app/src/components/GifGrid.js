import React from 'react';
import Proptypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { categoria } ) => {

    // const [images, setImages] = useState([]);
    
    const { data: images, loading } = useFetchGifs( categoria );

    // useEffect( () => {
    //     getGifs( categoria ).then( imgs => setImages( imgs )); //al modularizar getGifs, hubo que convertirlo en async y por ende en promesa
    // }, [ categoria ]); //dependencia a la categoria: si cambia la categoria vuelve a hacer la peticion a la api

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">"{ categoria }"</h3>

            { loading && <p className="animate__animated animate__flash">Cargando</p> }
            
            <div className="card-grid">
                {
                    images.map( img =>
                        <GifGridItem 
                            key={img.id}

                            // img={ img }
                            { ...img }
                        />
                    )
                }
            </div>
        </>
    )
};

GifGrid.proptypes = {
    categoria: Proptypes.string.isRequired
};

// SIN DESTRUCTURING

// <div>
// <h3>{ category }</h3>
// <ol>
//     {
//         images.map( image =>
//             <li key={image.id}>{image.title}</li>)
//     }
// </ol>
// </div
