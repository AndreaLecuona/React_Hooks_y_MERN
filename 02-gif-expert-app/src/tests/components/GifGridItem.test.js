import React from 'react';
import { GifGridItem } from "../../components/GifGridItem";
import {shallow} from 'enzyme';


describe('Pruebas en componente GifGridItem', () => {

    const title = 'Ejemplo';
    const url = 'https://images.app.goo.gl/jsTogSW8gkoT9zK46';

    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );


    test('Debe mostrar el componente GifGridItem correctamente', () => {

        const wrapper = shallow( <GifGridItem /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe renderizar GifGridItem junto a sus props correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('El componente debe contener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );

    });

    test('El componente debe contener una img con la url y una alt con el title', () => {
        
        const img = wrapper.find('img');
        //console.log( img.html() ); //podemos ver contenido y atributos del elemento
        //console.log( img.props() ); //equivalente pero en idioma react. Muestra las props como objetos
        
        //para ver una prop especifica
        //console.log( img.props().src ); //muestra solo el link
        //console.log( img.prop('src') ); //muestra el img como objeto con propiedades src y alt

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('El div principal del componente debe tener la clase animate__fadeIn', () => {
        
        const div = wrapper.find('div');

        //console.log( div.prop('className') ); //muestra todas las clases que tiene

        expect( div.hasClass('animate__fadeIn') ).toBe( true );

    });
    

});