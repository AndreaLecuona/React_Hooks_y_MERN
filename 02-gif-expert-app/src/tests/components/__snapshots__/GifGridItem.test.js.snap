// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem Debe mostrar el componente GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p>
     
     
  </p>
</div>
`;

exports[`Pruebas en componente GifGridItem Debe renderizar GifGridItem junto a sus props correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Ejemplo"
    src="https://images.app.goo.gl/jsTogSW8gkoT9zK46"
  />
  <p>
     
    Ejemplo
     
  </p>
</div>
`;
